arch/posix/eventloop_posix_tcp.c

```diff
@@ -231,6 +252,19 @@ TCP_connectionSocketCallback(UA_ConnectionManager *cm, TCP_FD *conn,
                         &UA_KEYVALUEMAP_NULL, response);
 }
 
+static void *
+removeListenSockets(void *application, UA_RegisteredFD *rfd) {
```

Comment:
You also need to make sure that new server sockets are not getting registered...